### DISCUSSION TITLE
Building/linking with libstdc++ debug symbols?

### DISCUSSION THREAD

**Discussion by shabble** [xpack-dev-tools/arm-none-eabi-gcc-xpack#20] [Q&A]
When building a c++ application (targetting an STM32F746) using `arm-none-eabi-gcc` from the xpack 10.3.1-2.3.1, it links to the `libstdc++.a` archive inside the package, and runs correctly.

The problem arises when attaching gdb, when examining some `std::stringstream` instances, they are reported as `<incomplete type>` which is apparently (partly?) related to not having debug symbols for the stdc++ lib.

I haven't yet found a way to determine if the symbols are in the package (in the abovementioned `.a` archive, or elsewhere) somewhere and there's some way to activate them, or if I would need to rebuild the whole libstdc++ with different flags, and if so, how to actually do that 😄

Any suggestions/pointers?

**Comment by ilg-ul** [comment]
Due to archive size constraints, the debug info is stripped from the released libraries.

I don't know of any simple trick to overcome this. You can try to bring the desired source files into your project and compile them together with the project.
